test/integration/select_test.lua

```diff
@@ -1508,3 +1508,41 @@ pgroup.test_jsonpath_index_field_pagination = function(g)
     local objects = crud.unflatten_rows(result.rows, result.metadata)
     t.assert_equals(objects, helpers.get_objects_by_idxs(cars, {2, 3}))
 end
+
+pgroup.test_select_timeout = function(g)
+    local customers = helpers.insert_objects(g, 'customers', {
+        {
+            id = 1, name = "Elizabeth", last_name = "Jackson",
+            age = 12, city = "New York",
+        }, {
+            id = 2, name = "Mary", last_name = "Brown",
+            age = 46, city = "Los Angeles",
+        }, {
+            id = 3, name = "David", last_name = "Smith",
+            age = 33, city = "Los Angeles",
+        }, {
+            id = 4, name = "William", last_name = "White",
+            age = 81, city = "Chicago",
+        },
+    })
+
+    table.sort(customers, function(obj1, obj2) return obj1.id < obj2.id end)
+
+    local result, err = g.cluster.main_server.net_box:call('crud.select', {'customers', nil, {timeout = 1}})
+
+    t.assert_equals(err, nil)
+    t.assert_equals(result.rows, {
+        {1, 477, "Elizabeth", "Jackson", 12, "New York"},
+        {2, 401, "Mary", "Brown", 46, "Los Angeles"},
+        {3, 2804, "David", "Smith", 33, "Los Angeles"},
+        {4, 1161, "William", "White", 81, "Chicago"},
+    })
+    t.assert_equals(result.metadata, {
+        {name = 'id', type = 'unsigned'},
+        {name = 'bucket_id', type = 'unsigned'},
+        {name = 'name', type = 'string'},
+        {name = 'last_name', type = 'string'},
+        {name = 'age', type = 'number'},
+        {name = 'city', type = 'string'},
+    })
```

Comment:
Is this check really necessary? (metadata)

Reply:
Removed.